pkg/agent/multicast/mcast_controller.go

```diff
@@ -183,28 +185,31 @@ func (c *Controller) clearStaleGroups() {
 	}
 }
 
-// removeLocalInterface searches the GroupMemberStatus which the deleted interface has joined, and then triggers a member
-// leave event so that Antrea can remove the corresponding interface from local multicast receivers on OVS. This function
-// should be called if the removed Pod receiver fails to send IGMP leave message before deletion.
-func (c *Controller) removeLocalInterface(e interface{}) {
+func (c *Controller) processPodUpdate(e interface{}) {
 	podEvent := e.(types.PodUpdate)
-	// Ignore Pod creation event.
-	if podEvent.IsAdd {
-		return
-	}
-	interfaceName := util.GenerateContainerInterfaceName(podEvent.PodName, podEvent.PodNamespace, podEvent.ContainerID)
-	ifConfig := &interfacestore.InterfaceConfig{
-		InterfaceName: interfaceName,
-	}
-	groupStatuses := c.getGroupMemberStatusesByPod(interfaceName)
-	for _, g := range groupStatuses {
-		event := &mcastGroupEvent{
-			group: g.group,
-			eType: groupLeave,
-			time:  time.Now(),
-			iface: ifConfig,
+	if !podEvent.IsAdd {
+		// Searches the GroupMemberStatus which the deleted interface has joined, and then triggers a member
+		// leave event so that Antrea can remove the corresponding interface from local multicast receivers on OVS. This function
+		// should be called if the removed Pod receiver fails to send IGMP leave message before deletion.
+		interfaceName := util.GenerateContainerInterfaceName(podEvent.PodName, podEvent.PodNamespace, podEvent.ContainerID)
+		ifConfig := &interfacestore.InterfaceConfig{
+			InterfaceName: interfaceName,
+		}
+		groupStatuses := c.getGroupMemberStatusesByPod(interfaceName)
+		for _, g := range groupStatuses {
+			event := &mcastGroupEvent{
+				group: g.group,
+				eType: groupLeave,
+				time:  time.Now(),
+				iface: ifConfig,
+			}
+			c.groupEventCh <- event
 		}
-		c.groupEventCh <- event
+	}
+	// Update flows MulticastEgressPodMetricTables when pod interfaces change.
+	err := c.installMulticastEgressPodFlow()
```

Comment:
I don't think it is a good idea to install multicast egress pod metric flow for all local Pods: 1) It is a waste of flow entries if the Pod never sends Multicast traffic, 2) those Pod never sends multicast traffic is also counted in the metric result per pod, which is meaning less. 3) we don't know the multicast group address at this time, so the flow can only match a destination with 224.0.0.0/4, but it is meaningless to collect the counters a Pod sends traffic to all Multicast address.

So I would prefer to install Egress pod metric flow only when a Pod is applied to an egress rule on Multicast traffic, in this case, we could reduce the useless flow entris, and we could focus on the exact Multicast address. Besides, it is better to call the flow installation in NetworkPolicy module in this case.

Reply:
I do not have a solid solution to solve the "install flows for all local Pods" problem as you mentioned. Let say for a node that has 100 local pods, installing 100 flow entries in  MulticastEgressPodFlowTable should not be a huge overhead. I will keep this implementation in this pr and may find to way to optimize it in future. 

For the suggested implementation. I prefer not binding the pod-level multicast stats to networkpolicy.